### DISCUSSION TITLE
Creating custom modifier calculation

### DISCUSSION THREAD

**Discussion by Solanaar** [Obsidian-TTRPG-Community/fantasy-statblocks#319] [Q&A]
Hello everyone. I am trying to create a numenera vault and am currently attempting to create a custom statblock template.

Numenera is extremely simple. Few calculations. Actually only 1: Target number. it's literally level * 3. I know I could just do this in my head but I like to style the stat blocks like they are in the source books. There it is written as "level (target number)", e.g. 5 (15). In the default layout stats are styled in a similar way so I thought I can achieve the same...

I am incredibly overwhelmed by almost everything. I am not a programmer and almost anything obsidian is gibberish to me (and I understand basic c++ somewhat). But I feel like programmers write documentation very differently and I struggle a lot in deciphering seemingly simple sentences. So please excuse if some of these questions seem incredibly obvious to you.

So first of all: what language am I supposed to use for dice callbacks, ability modifiers, etc? I found mention of YAML and JavaScript in the documentation. Does that mean I need to learn these to be able to create custom statblocks? I studied the provided default layout and tried to reverse engineer a few things, but it takes a lot of time and I understand very little of what I'm doing so I am unable to troubleshoot stuff when it ultimately doesn't work the way I expected.

So to summarize what I've got so far:
I need to **Add** a new **Table** element, I need to write "level" into the **Link Monster Property** field, cause when I write stuff into the statblock in the edit view I need to write down "level: number" (and obv. write a specific number for actual stat blocks). Then I need to check **Calculate Modifiers** and write _something_ into the last field.
What I don't know is: do I need to provide a table header if I'll only use one "stat" value?
And finally: what exactly do I write into the last field so it displays the target number next to the level?

I hope this makes a semblance of sense to anyone and I'm glad for any help. Also please excuse the frustrated undertones, I am, well, frustrated, by my experiences with Obsidian and all those plugins so far. It feels like every single function needs a 2-3 hour course just to be able to use it, if not more because you'll need to know html, css, yaml, javascript, etc.

**Comment by mbound** [comment]
I am also trying to figure out similar things, so maybe we can brainstorm together.  To clarify, when you say modifier are you referring to the Initiative Modifier? Or are you simply trying to derive the `Target` number from the `level` with a formula?

**Reply by Solanaar** [reply]
The latter, iirc. Though, to be honest, I dropped statblock for multiple reasons and resorted to formatting statblocks in markdown and manually, with some css styling, though I try to keep that to a minimum.

**Comment by TrindadeBr** [comment]
i'm doing a character sheet for 5e, and i know a little bit about javascript and very little html, and i am having difficulties as well. BUT i found out a few things. 

The languages: javascript and html. thank god. still, there is a few caveats, since the result has to aways be a string, and you can deliver a code, so long it obeys the string form. <- this took a beat out of me on the first days. beating my head on the keyboard till a result was found on one of my formulae. 

the callbacks: there are two types of callbacks fields - dice callbacks and callbacks. if dice callbacks are provided, the ''normal'' callback will not be listened/read. 
while there is a example of a callback on the very same window of it, heres one that i'm using , just to quickly run a d20 next to a field : 
`const dice = "d20";
return [{ text: dice }];`
if i want to add a modifier, lets say strenght, and add proficiency/expertise bonus, i would first have to calculate the modifier. There are several modifications to how i organized my statblock, and how i had to approach the code that allow me to do so. But in the end, here is the code :

`let proList = [[monster.savingThrowProficiences]]+"";`
`let proBonus = Math.floor(([monster.level]-1)/4)+2; `
`let mod = Math.round(([[monster.stats[0]]]-11)/2); `
`if(proList.includes("Strenght")) {`
`    mod+=proBonus `
`} `
`let dice = mod +"+ d20";`
`return [{ text: dice }]  `

proList function is getting a var on the statblock, adding quotes to make it into a string ( because if i make a table on the statblock, it cannot find the thing i am looking for. refer "beating my head at keyboard" in the text above ) 

proBonus function is calculating proficiency bonus based off statblock level. the formulae works pretty well, and i dont have to map into a list.

mod function is calculating the modifier. round(stat -10/2) results are wrong, so -11 it is. works like a charm. Also, since it works like a js List, you can modify the [0] (strenght) , to [5] (charisma)

the if is check if there is said proficiency, and adding proBonus to mod... += line  is equivalent to ` mod = mod + proBonus`
in the end, same as dice code before, but add the mod. 
note that the code will deliver a string, so the result is returning (if strenght 12) "1+ d20". remember to toggle the option "parse for dice " option on the same window you add the code.

The thing is, you have to copy paste this ( and modify to the stat you want ) every single saving throw. And this is just saving throw. and i'm not considering features, like alert would add perception etc.

sorry for the necromantic, huge reply, and for english mistakes. and for bad code. but if it works....
